src/viur/core/modules/file.py

```diff
@@ -869,6 +944,106 @@ def download(self, blobKey: str, fileName: str = "", download: bool = False, sig
 
         raise errors.Redirect(signedUrl)
 
+    @exposed
+    def serve(
+        self,
+        key: str,
+        size: t.Optional[int] = None,
+        filename: t.Optional[str] = None,  # random string with .ext
+        options: str = "",
+        download: bool = False,
+    ):
+        """
+        Requests an image using the serving url to bypass direct Google requests.
+
+        :param key: a string with one __ seperator. The first part is the host prefix, the second part is the key
+        :param size: the target image size, take a look at the VALID_SIZES
+        :param filename: a random string with an extention, valid extentions are jpg,png,webp
+        :param options: - seperated options:
+            c - crop
+            p - face crop
+            fv - vertrical flip
+            fh - horizontal flip
+            rXXX - rotate 90, 180, 270
+            nu - no upscale
+        :param download: if value = 1 force download, else not
+        :return:
+        """
+
+        VALID_OPTIONS = {
```

Comment:
Why aren't these constants class variables, which would allow monkeypatching / subclasse overwriting for project requirements?

Reply:
Constants moved to `File`-class.